components/connected_components.py

```diff
@@ -0,0 +1,81 @@
+# MIT License
+#
+# Copyright (c) 2021 DNS Technology
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""Connected components.
+
+Note on optimising TensorFlow GPU performance:
+ 1. https://www.tensorflow.org/guide/gpu_performance_analysis
+ 2. https://github.com/NVIDIA/DeepLearningExamples/issues/57
+"""
+
+import tensorflow as tf
+import tensorflow_addons as tfa
+import numpy as np
+import os
+
+os.environ['TF_GPU_THREAD_MODE'] = 'gpu_private'
```

Comment:
@arvindrajan92 is this necessary for ConnectedComponents? Or is this left over from testing?

Reply:
There is some performance gain with setting the environment variable for TensorFlow: https://www.tensorflow.org/guide/gpu_performance_analysis. So I left it there.